workspaces/cli-shared/src/captures/avro/file-system/capture-saver.ts

```diff
@@ -101,13 +109,39 @@ export class CaptureSaver implements ICaptureSaver {
   }
 
   async save(sample: IHttpInteraction) {
+    this.interactionsReceivedCount++;
     // don't await flush, just enqueue
-    await this.batcher.add(sample);
+    this.batcher.add(sample);
   }
 
   async cleanup() {
     developerDebugLogger('stopping capture saver');
+
+    await new Promise((resolve, reject) => {
+      const poll = () => {
+        const interactionsReceivedCount = this.interactionsReceivedCount;
+        const interactionsSavedCount = this.interactionsSavedCount;
+        developerDebugLogger(
+          'waiting until interactionsReceivedCount matches interactionsSavedCount...',
+          interactionsReceivedCount,
+          interactionsSavedCount
+        );
+        if (interactionsSavedCount === interactionsReceivedCount) {
```

Comment:
I'd consider `>=` here, from a fault-tolerance perspective in case there's some bug in arithmetic, but I might be just a bit too paranoid there.

Reply:
I think if somehow the saved count advanced past the received count, we'd have some interesting issues at hand